src/main/java/org/wso2/carbon/connector/KafkaProduceConnector.java

```diff
@@ -442,6 +440,14 @@ private Object handleComplexAvroTypes(Object jsonString, Schema schema) {
         boolean usingDefault = false;
         switch (schema.getType()) {
             case BYTES:
+                if ("decimal".equals(schema.getLogicalType().getName())) {
```

Comment:
Shall we define a constant for `decimal`?

Reply:
Fixed